v2/v2builder/v2builder.go

```diff
@@ -357,15 +357,15 @@ func (i BuilderImpl) GenUserFacing(ctx context.Context, p builder.GenUserFacingP
 				// Service structs are not needed if there is no implementation to be generated
 				svcStruct := option.None[*codegen.VarDecl]()
 
+				buf.Reset()
```

Comment:
You introduced a bug in v2 which has been iterating me all year ;) 

Without the reset, it causes us to generate empty `encore.gen.go` files when they are not needed. (As you unconditionally always add the prelude comments). I explicly didn't do this in v1, and this fixes it in v2.

Reply:
Ah nice. We also have an outstanding issue where we keep `encore.gen.go` files around after a service has been renamed. Think that would be easy to address, too, or is that a separate issue?

Reply:
Seperate issue, mainly because we don't track services in the CLI so we have no way of detecting that easily (apart from maybe looking for `encore.gen.go` files being the only thing in a folder - but I say let's keep that as a separate issue